File: templates/go-service/service/example/components/greeting/eventHandler.go
package greeting

import (
	"app/context"
	"app/transport"
)

// CreateRegisterPersonHandler create event handler for "registerPerson"
func CreateRegisterPersonHandler(ctx *context.Context) transport.EventHandler {
	return func(msg transport.Message) (err error) {
		name, err := msg.GetString("name")
		if err != nil {
			return err
		}
		ctx.InitLocalCache("names", []string{})
		if name != "" {
			names, err := ctx.LocalCache.GetStringArray("names")
			if err != nil {
				return err
			}
			ctx.LocalCache["names"] = append(names, name)
		}
		return err
	}
}

File: templates/nodejs-service/service/src/components/example/eventHandler.ts
import { Context } from "../../context";
import { EventHandler, Message } from "../../transport";

export function createRegisterPersonHandler(ctx: Context): EventHandler {
    return (msg: Message) => {
        const { name } = msg;
        ctx.initLocalCache("names", []);
        if (name != "") {
            const { names } = ctx.localCache;
            names.push(name);
            ctx.localCache["names"] = names;
        }
    }
}